### ISSUE TITLE
Add support for Geocoding (Address Lookups)

### ISSUE THREAD

**Issue by MoritzLost** [doublesecretagency/craft-mapbox#8]
I need to import some entries with an address field where the source data only includes a full address, not separate fields for street, city, zip code etc. I can do this manually through the API, but that's pretty bothersome. It would be nice if the plugin came with a utility function that performs an address lookup and returns an `Address` model that I can just plug into the address field.

**Comment by lindseydiloreto** [comment]
This is a good feature request, thanks. You're roughly describing **geocoding**, which will be critical for the plugin long-term.

Geocoding support is implied (but not stated) in issue #1. It's a fairly high priority for us, although we're working it into a busy schedule. With any luck, this functionality will exist within the next month or so. 🤞

**Comment by MoritzLost** [comment]
@lindseydiloreto Yes, in principle it's geocoding – though some geocoding APIs return only coordinates, not structured address data, if I recall correctly. For our current use-case we'd need to be able to give it an address as a single string and get back both coordinates and structured address data (street & number, zip code, city, etc).

**Comment by MoritzLost** [comment]
@lindseydiloreto Any update on this? We have a project where we will need geocoding soon …
